src/main/groovy/com/morpheusdata/commvault/backup/openstack/CommvaultOpenstackBackupTypeProvider.groovy

```diff
@@ -0,0 +1,214 @@
+	 * NOTE: This may be useful to set as an i18n key for UI reference and localization support.
+	 *
+	 * @return either an English name of a Provider or an i18n based key that can be scanned for in a properties file.
+	 */
+	@Override
+	String getName() {
+		return "Commvault OpenStack Backup"
+	}
+	
+	/**
+	 * get the type of container compatible with this backup type
+	 * @return the container type code
+	 */
+	@Override
+	String getContainerType() {
+		return "single"
+	}
+	
+	/**
+	 * Determines if this backup type supports copying the backup to a datastore (export).
+	 * @return boolean indicating copy to store is supported
+	 */
+	@Override
+	Boolean getCopyToStore() {
+		return false
+	}
+
+	/**
+	 * Determines if this backup type supports downloading the backups for this backup type.
+	 * @return boolean indicating downloading is enabled
+	 */
+	@Override
+	Boolean getDownloadEnabled() {
+		return false
+	}
+
+	/**
+	 * Determines if this backup type supports restoring a backup to the existing workload.
+	 * @return boolean indicating restore to existing is enabled
+	 */
+	@Override
+	Boolean getRestoreExistingEnabled() {
+		return true
+	}
+
+	/**
+	 * Determines if this backup type supports restoring to a new workload rather than replacing the existing workload.
+	 * @return boolean indicating restore to new is enabled
+	 */
+	@Override
+	Boolean getRestoreNewEnabled() {
+		return true
+	}
+
+	/**
+	 * Indicates the type of restore supported. Current options include: new, existing, online, offline, migration, failover
+	 * @return the supported restore type
+	 */
+	@Override
+	String getRestoreType() {
+		return "online"
+	}
+
+	/**
+	 * Get the desired method of restoring a backup to a new instance.
+	 * <p>
+	 * Available options:
+	 * 		<ul>
+	 * 			<li>
+	 * 			 	DEFAULT -- Uses the backup as an input to the instance provision process. Generally this
+	 * 			 				involves using a snapshot or image ID as the image used for provisioning, but the precise
+	 * 			 				details are left up to the backup provider.
+	 * 			</li>
+	 * 		 	<li>
+	 * 		 	    VM_RESTORE -- The external backup provider restores to a VM and the core system will associate the resulting
+	 * 		 	    				VM to the internal resources.
+	 * 		 	</li>
+	 * 		 	<li>
+	 * 		 	  	TEMP_EXTRACT -- determines the visibility of the restore to new option
+	 * 		 	</li>
+	 * 		</ul>
+	 *
+	 * @return
+	 */
+	@Override
+	String getRestoreNewMode() {
+		return "VM_RESTORE"
+	}
+	
+	/**
+	 * Does this backup type provider support copy to store?
+	 * @return boolean indicating copy to store support
+	 */
+	@Override
+	Boolean getHasCopyToStore() {
+		return false
+	}
+
+	/**
+	 * A list of {@link OptionType OptionTypes} for use in the backup create and edit forms.
+	 * @return a list of option types
+	 */
+	@Override
+	Collection<OptionType> getOptionTypes() {
+		return new ArrayList<OptionType>()
+	}
+
+	BackupTypeProvider getBackupTypeProvider() {
```

Comment:
Lines 151 to 165 appear to be Veeam specific methods that have been copied over to commvault. Please remove if they are not being used. Clean up the other backup type providers as well.

Reply:
@ddeyoung changes are made as suggested